devicetypes/timevalve-gaslock-t-08/timevalve-smart.src/timevalve-smart.groovy

```diff
@@ -0,0 +1,281 @@
+	def result = []
+	log.info "zwave.configurationV1.configurationGet - " + cmd
+    def array = cmd.configurationValue
+    def value = ( (array[0] * 0x1000000) + (array[1] * 0x10000) + (array[2] * 0x100) + array[3] ).intdiv(60)
+    if (device.currentValue("switch") == "on") {
+    	value = ( (array[0] * 0x1000000) + (array[1] * 0x10000) + (array[2] * 0x100) + array[3] ).intdiv(60)
+    } else {
+    	value = 0
+    }
+    
+    if (device.currentValue('contact') == 'open') {
+    	def hour = value.intdiv(60);
+        def min = (value % 60).toString().padLeft(2, '0');
+        def text = "${hour}:${min}M"
+        
+        log.info "remain - " + text
+        result.add( createEvent(name: "remaining", value: value, displayed: false, isStateChange: true) )
+        result.add( createEvent(name: "remainingText", value: text, displayed: false, isStateChange: true) )
+    } else {
+    	result.add( createEvent(name: "timeout", value: value, displayed: false, isStateChange: true) )
+    }
+    
+    /*
+    if (cmd.parameterNumber == 0x01) { // timeout
+    	result.add( createEvent(name: "timeout", value: value, displayed: false, isStateChange: true) )
+        
+        
+        // timeout 설정 시에도 
+        def hour = value.intdiv(60);
+        def min = (value % 60).toString().padLeft(2, '0');
+        def text = "${hour}:${min}M"
+        
+        result.add( createEvent(name: "remaining", value: value, displayed: false, isStateChange: true) )
+        result.add( createEvent(name: "remainingText", value: text, displayed: false, isStateChange: true) )
+    } else if (cmd.parameterNumber == 0x03) { // remaining
+        def hour = value.intdiv(60);
+        def min = (value % 60).toString().padLeft(2, '0');
+        def text = "${hour}:${min}M"
+        
+        debug.info "remain - " + text
+        result.add( createEvent(name: "remaining", value: value, displayed: false, isStateChange: true) )
+        result.add( createEvent(name: "remainingText", value: text, displayed: false, isStateChange: true) )
+    }
+    */
+    return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	def type = cmd.notificationType
+    if (type == cmd.NOTIFICATION_TYPE_HEAT) {
+    	log.info "NotificationReport - ${type}"
+        createEvent(name: "temperature", value: 80, descriptionText: "${device.displayName} is over heat!", displayed: true, isStateChange: true)
+    }
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.alarmv1.AlarmReport cmd) {
+	def type = cmd.alarmType
+    def level = cmd.alarmLevel
+    
+    log.info "AlarmReport - type : ${type}, level : ${level}"
+    def msg = "${device.displayName} is over heat!"
+    def result = createEvent(name: "temperature", value: 80, descriptionText: msg, displayed: true, isStateChange: true)
+    if (sendPushMessage) {
+    	sendPushMessage(msg)
+    }
+    return result
+}
+
+// remote open not allow
```

Comment:
If open is not allowed to be supported for remote control, then the open() function should be empty. Otherwise, SmartApps will be able to call open() on the device and execute open().